internal/services/apimanagement/api_management_logger_resource.go

```diff
@@ -256,12 +275,29 @@ func resourceApiManagementLoggerDelete(d *pluginsdk.ResourceData, meta interface
 	return nil
 }
 
-func expandApiManagementLoggerEventHub(input []interface{}) *map[string]string {
+func expandApiManagementLoggerEventHub(input []interface{}) (*map[string]string, error) {
 	credentials := make(map[string]string)
 	eventHub := input[0].(map[string]interface{})
+
+	connectionString := eventHub["connection_string"].(string)
+	endpointAddress := eventHub["endpoint_address"].(string)
+	clientId := eventHub["client_id"].(string)
+	if len(eventHub["connection_string"].(string)) == 0 && len(eventHub["endpoint_address"].(string)) == 0 {
+		return nil, fmt.Errorf("Either `connection_string` or `endpoint_address` is required")
+	}
```

Comment:
This validation can actually be done up in the schema at plan time by specifying `AtLeastOneOf`
```suggestion
```